### PR TITLE
Yvesdup queue shutdown changes

### PR DESCRIPTION
**Pull request by EpicWink**

### CODE REVIEW

File: Lib/asyncio/queues.py [hunk 1]
```diff
@@ -200,7 +202,7 @@ async def get(self):
                     # the call.  Wake up the next in line.
                     self._wakeup_next(self._getters)
                 raise
-            if self._shutdown_state == _QueueState.shutdown_immediate:
+            if self._shutdown_state is not _QueueState.alive:
```
Comment (EpicWink) [inline Lib/asyncio/queues.py]: @YvesDup what is the reason for this? `_shutdown_state is not _QueueState.alive` will be tested in the next iteration of the `while` loop: we only need to test if the queue is in the shutdown-immediate state

File: Lib/multiprocessing/queues.py [hunk 2]
```diff
@@ -159,6 +161,9 @@ def get_nowait(self):
     def put_nowait(self, obj):
         return self.put(obj, False)
 
+    def shutdown(self, immediate=True):
+        pass
+
```
Comment (EpicWink) [inline Lib/multiprocessing/queues.py]:
```suggestion
```

Already implemented below

File: Lib/queue.py [hunk 3]
```diff
@@ -29,9 +30,10 @@ class ShutDown(Exception):
     '''Raised when put/get with shut-down queue.'''
 
 
-_queue_alive = "alive"
-_queue_shutdown = "shutdown"
-_queue_shutdown_immediate = "shutdown-immediate"
+class _QueueState(enum.Enum):
+    ALIVE = "alive"
+    SHUTDOWN = "shutdown"
+    SHUTDOWN_IMMEDIATE = "shutdown-immediate"
```
Comment (EpicWink) [inline Lib/queue.py:36]: Do we want to introduce a new `enum` import into threading `queue`? It doesn't currently import it.

**Review by EpicWink — COMMENTED** [review]
@YvesDup your changes reviewed

File: Lib/queue.py (hunk 3 above)
Comment (YvesDup) [reply Lib/queue.py:36]: Yes, I would in order to be consistent with asyncio lib

File: Lib/multiprocessing/queues.py (hunk 2 above)
Comment (YvesDup) [reply Lib/multiprocessing/queues.py]: Yes I did it.

File: Lib/queue.py (hunk 3 above)
Comment (YvesDup) [reply Lib/queue.py:36]: There is a comment in your main PR. So this proposition is rejected.

File: Lib/asyncio/queues.py (hunk 1 above)
Comment (YvesDup) [reply Lib/asyncio/queues.py]: It is a mistake. Ok

**Pull request closed without merging**